log.go

```diff
@@ -48,52 +43,79 @@ import (
-	sgnrLog = build.NewSubLogger("SGNR", backendLog.Logger)
-	wlktLog = build.NewSubLogger("WLKT", backendLog.Logger)
-	arpcLog = build.NewSubLogger("ARPC", backendLog.Logger)
-	invcLog = build.NewSubLogger("INVC", backendLog.Logger)
-	nannLog = build.NewSubLogger("NANN", backendLog.Logger)
-	wtwrLog = build.NewSubLogger("WTWR", backendLog.Logger)
-	ntfrLog = build.NewSubLogger("NTFR", backendLog.Logger)
-	irpcLog = build.NewSubLogger("IRPC", backendLog.Logger)
-	chnfLog = build.NewSubLogger("CHNF", backendLog.Logger)
-	chbuLog = build.NewSubLogger("CHBU", backendLog.Logger)
-	promLog = build.NewSubLogger("PROM", backendLog.Logger)
-	wtclLog = build.NewSubLogger("WTCL", backendLog.Logger)
-	prnfLog = build.NewSubLogger("PRNF", backendLog.Logger)
+	logWriter = build.NewRotatingLogWriter()
+
+	ltndLog = build.NewSubLogger("LTND", logWriter.GenSubLogger)
+	lnwlLog = build.NewSubLogger("LNWL", logWriter.GenSubLogger)
+	peerLog = build.NewSubLogger("PEER", logWriter.GenSubLogger)
+	discLog = build.NewSubLogger("DISC", logWriter.GenSubLogger)
+	rpcsLog = build.NewSubLogger("RPCS", logWriter.GenSubLogger)
+	srvrLog = build.NewSubLogger("SRVR", logWriter.GenSubLogger)
+	ntfnLog = build.NewSubLogger("NTFN", logWriter.GenSubLogger)
+	chdbLog = build.NewSubLogger("CHDB", logWriter.GenSubLogger)
+	fndgLog = build.NewSubLogger("FNDG", logWriter.GenSubLogger)
+	hswcLog = build.NewSubLogger("HSWC", logWriter.GenSubLogger)
+	utxnLog = build.NewSubLogger("UTXN", logWriter.GenSubLogger)
+	brarLog = build.NewSubLogger("BRAR", logWriter.GenSubLogger)
+	cmgrLog = build.NewSubLogger("CMGR", logWriter.GenSubLogger)
+	crtrLog = build.NewSubLogger("CRTR", logWriter.GenSubLogger)
+	btcnLog = build.NewSubLogger("BTCN", logWriter.GenSubLogger)
+	atplLog = build.NewSubLogger("ATPL", logWriter.GenSubLogger)
+	cnctLog = build.NewSubLogger("CNCT", logWriter.GenSubLogger)
+	sphxLog = build.NewSubLogger("SPHX", logWriter.GenSubLogger)
+	swprLog = build.NewSubLogger("SWPR", logWriter.GenSubLogger)
+	sgnrLog = build.NewSubLogger("SGNR", logWriter.GenSubLogger)
+	wlktLog = build.NewSubLogger("WLKT", logWriter.GenSubLogger)
+	arpcLog = build.NewSubLogger("ARPC", logWriter.GenSubLogger)
+	invcLog = build.NewSubLogger("INVC", logWriter.GenSubLogger)
+	nannLog = build.NewSubLogger("NANN", logWriter.GenSubLogger)
+	wtwrLog = build.NewSubLogger("WTWR", logWriter.GenSubLogger)
+	ntfrLog = build.NewSubLogger("NTFR", logWriter.GenSubLogger)
+	irpcLog = build.NewSubLogger("IRPC", logWriter.GenSubLogger)
+	chnfLog = build.NewSubLogger("CHNF", logWriter.GenSubLogger)
+	chbuLog = build.NewSubLogger("CHBU", logWriter.GenSubLogger)
+	promLog = build.NewSubLogger("PROM", logWriter.GenSubLogger)
+	wtclLog = build.NewSubLogger("WTCL", logWriter.GenSubLogger)
+	prnfLog = build.NewSubLogger("PRNF", logWriter.GenSubLogger)
+
+	// subsystemLoggers maps each subsystem identifier to its associated
+	// logger.
+	subsystemLoggers = build.SubLoggers{
+		"LTND": ltndLog,
```

Comment:
can we build this map by doing something like:

```go
build.SubLoggers{
    ltndlog.Abbr(): ltndLog,
    ...
}
```

Reply:
I did some more refactoring and now it's even more compact.